modin/pandas/test/test_groupby.py

```diff
@@ -1714,3 +1766,110 @@ def test_not_str_by(by, as_index):
     eval_general(md_grp, pd_grp, lambda grp: grp.agg(lambda df: df.mean()))
     eval_general(md_grp, pd_grp, lambda grp: grp.dtypes)
     eval_general(md_grp, pd_grp, lambda grp: grp.first())
+
+
+@pytest.mark.parametrize("internal_by_length", [0, 1, 2])
+@pytest.mark.parametrize("external_by_length", [0, 1, 2])
+@pytest.mark.parametrize("has_categorical_by", [True, False])
+@pytest.mark.parametrize(
+    "agg_func",
+    [
+        pytest.param(
+            lambda grp: grp.apply(lambda df: df.dtypes), id="modin_dtype_impl"
+        ),
+        pytest.param(lambda grp: grp.apply(lambda df: df.sum()), id="apply(sum)"),
+        pytest.param(lambda grp: grp.count(), id="count"),
+        pytest.param(lambda grp: grp.nunique(), id="nunique"),
+        pytest.param(
+            {"c": "sum", "d": "nunique"}, id="dict_agg_no_intersection_with_by"
+        ),
+        pytest.param(
+            {"b": "mean", "c": "sum", "d": "nunique"},
+            id="dict_agg_has_intersection_with_by",
+        ),
+        pytest.param(
+            {"a": "nunique", "c": "sum", "d": "nunique"},
+            id="dict_agg_has_intersection_with_categorical_by",
+        ),
+    ],
+)
+def test_handle_as_index(
+    internal_by_length, external_by_length, has_categorical_by, agg_func, request
+):
+    """
+    Test ``GroupBy.handle_as_index``.
+
+    The role of the ``handle_as_index`` method is to build a groupby result considering
+    ``as_index=False`` from the result that was computed with ``as_index=True``.
+
+    So the testing flow is the following:
+        1. Compute GroupBy result with the ``as_index=True`` parameter via Modin.
+        2. Build ``as_index=False`` result from the ``as_index=True`` using ``handle_as_index`` method.
+        3. Compute GroupBy result with the ``as_index=False`` parameter via Pandas (is the reference result).
+        4. Compare the result from the second step with the reference.
+    """
+    by_length = sum((internal_by_length, external_by_length))
+    if by_length == 0:
+        return
```

Comment:
shouldn't this be an assertion that the function isn't actually testing anything (called uselessly)?

Reply:
you mean `pytest.skip` or something similar?

added this for now:
```python
    if by_length == 0:
        pytest.skip("No keys to group on was passed, skipping the test.")
```